org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/core/ReplModule.java

```diff
@@ -25,9 +26,11 @@
 import com.google.inject.multibindings.MapBinder;
 
 /**
- * Client library bindings.
+ * This class binds the core classes. It is intended to be subclassed by client implementations.
+ * These subclasses should bind their implementations of {@link IRepl}, {@link IMessageHook} and
+ * {@link IResultHook}.
  */
-public class ReplModule extends SpoofaxModule {
+public abstract class ReplModule extends SpoofaxModule {
```

Comment:
In #29, the only dummy required would be an `IRepl` dummy. Note that a dummy can just be a lambda that prints an error, or better yet throws an exception so that Guice complains.